core/Symbols.h

```diff
@@ -142,7 +142,10 @@ class Symbol final {
 
     // Attempts to add the given mixin to the symbol. If the mixin is invalid because it is not a module, it returns
     // `false` (but still adds the mixin for processing during linearization) and the caller should report an error.
-    [[nodiscard]] bool addMixin(const GlobalState &gs, ClassOrModuleRef sym);
+    [[nodiscard]] bool addMixin(const GlobalState &gs, ClassOrModuleRef sym, std::optional<u2> index = std::nullopt);
+
+    // Add a placeholder for a mixin and return index in mixins()
+    u2 addStubMixin(const GlobalState &gs);
```

Comment:
minor thing: can we potentially use a different name? sorbet currently uses `stub` exclusively to describe the state when it had to synthesize a constant because the source referred to something that didn't exist in the symbol table (i.e., "Unable to resolve constant" errors)

Maybe placeholder?